hdump.c

```diff
@@ -90,20 +93,36 @@ int main(int argc, char *argv[])
 	if (!cols)
 		cols = 16;
 
-	buff = (unsigned char *) calloc (1, sizeof(unsigned char) * cols);
-	ascii = (unsigned char *) calloc (1, (sizeof(unsigned char) * cols) + 1);
```

Comment:
I moved buff and ascii allocations to after the tests about opening the file are done because I noticed that fatal() function doesn't free buff and ascii variables, I know that nowdays operatin systems already free allocated memory when a proccess is done, but I read somewhere that freeing through the process itself would be better so, here it goes.